src/globus_sdk/services/search/client.py

```diff
@@ -39,6 +39,119 @@ class SearchClient(client.BaseClient):
     # Index Management
     #
 
+    def create_index(
+        self, display_name: str, description: str
+    ) -> response.GlobusHTTPResponse:
+        """
+        Create a new index.
+
+        :param display_name: the name of the index
+        :type display_name: str
+        :param description: a description of the index
+        :type description: str
+
+        New indices default to trial status. For subscribers with a subscription ID,
+        indices can be converted to non-trial by sending a request to support@globus.org
+
+        .. tab-set::
+
+            .. tab-item:: Example Usage
+
+                .. code-block:: python
+
+                    sc = globus_sdk.SearchClient(...)
+                    r = sc.create_index(
+                        "History and Witchcraft",
+                        "Searchable information about history and witchcraft",
+                    )
+                    print(f"index ID: {r['id']}")
+
+            .. tab-item:: Example Response Data
+
+                .. expandtestfixture:: search.create_index
+
+            .. tab-item:: API Info
+
+                ``POST /v1/index``
+
+                .. extdoclink:: Index Create
+                    :ref: search/reference/index_create/
+        """
+        log.info(f"SearchClient.create_index({display_name!r}, ...)")
+        return self.post(
+            "/v1/index", data={"display_name": display_name, "description": description}
+        )
+
+    def delete_index(self, index_id: UUIDLike) -> response.GlobusHTTPResponse:
+        """
+        Mark an index for deletion.
+
+        Globus Search does not immediately delete indices. Instead, this API sets the
+        index status to ``"delete-pending"``.
+        Search will move pending tasks on the index to the ``CANCELLED`` state and will
+        eventually delete the index.
+
+        If the index is a trial index, it will be deleted a few minutes after being
+        marked for deletion.
+        If the index is non-trial, it will be kept for 30 days and will be eligible for
+        use with the ``reopen`` API (see :meth:`~.reopen_index`) during that time.
```

Comment:
This seems like the wrong place to document the Search service backend implementation and limits (like "30 days").

Reply:
Then perhaps only something shorter? I wanted to pull in something from the doc because `reopen` only makes sense with the broader context that `delete` may leave your index hanging out _but only for non-trial_.

I am a bit concerned that completely omitting the period (30 days) might make the doc here more confusing to read because it makes us describe it as something indefinite, when we do know the value. What about "several weeks"?

This is my crack at a rephrase:

> If the index is a trial index, it will be deleted shortly, but if it is non-trial, it will be kept in the ``"delete-pending"`` state for several weeks.
> Non-trial indices in the ``"delete-pending"`` state are eligible for use with the ``reopen`` API (see :meth:`~.reopen_index`) during that time.

Reply:
Hmm...is it possible to link to the documented constraints?

I like the specificity of "30 days" _et al_, but documenting the behavior and limits here seems duplicative, and worries me that it might age poorly. We can keep this as-is but I wanted to voice that concern.

Oh! I just realized I unlocked an achievement!

![image](https://github.com/globus/globus-sdk-python/assets/39996/77914792-cb6b-419b-a906-7247326e5a1a)

Reply:
The API reference doc link provides the requisite context, and since it's at the top of that page, there isn't really a better link to use than that one.

As of right now, I'm gently inclined to keep this as-is unless we have some alternative phrasing which still draws the connection that, for non-trial indices, `delete->reopen` is possible (for a time).

Reply:
I just saw that you approved, so I went ahead with the merge. But I'm happy to edit this further if there's a rephrase we can work out.